### PR TITLE
Updated ThreadTrackingStatistic to use PerfCounterTelemetryConsumer

### PR DESCRIPTION
**Pull request by galvesribeiro**
More work for #2122

### CODE REVIEW

**Comment by galvesribeiro** [comment]
CC: @jason-bragg / @jdom

File: src/Orleans/Statistics/ThreadTrackingStatistic.cs [hunk 1]
```diff
@@ -163,7 +162,7 @@ public void OnStartProcessing()
                 firstStart = false;
                 if (StatisticsCollector.CollectContextSwitchesStats)
                 {
-                    TrackContextSwitches();
+                    logger.IncrementMetric(metricName);
```
Comment (jason-bragg) [inline src/Orleans/Statistics/ThreadTrackingStatistic.cs]: I don't think this is a good way to trigger logging these metrics.  In general, all we are doing here is telling the telemetry that there is a metric of this name that is being incremented, which is not the case.  We are then relying on the Orleans perf counter metric consumer to detect this metric and perform special processing on it.  This is quite kludgy.

The problem is that we don't have anything to log here, we just need to signal something else to retrieve and log the data, so this is really a signal to an framework extension point.  I suspect that DI could help.

Suggest something like:
1) Create public IThreadTrackingStatistic interface with the public surface of ThreadTrackingStatistic.
2) Move ThreadTrackingStatistic to OrleansTelemetryConsumers.Counters as an implementation of IThreadTrackingStatistic.
3) Add extension function to register ThreadTrackingStatistic type in the service collection for DI.
4 Update Orleans to get IThreadTrackingStatistic form DI.

Comment (tsibelman) [inline src/Orleans/Statistics/ThreadTrackingStatistic.cs]: I second your coment what will happen if I implement another telemetry consumer ? And I will get just a single call to Context Switches/sec, what is it means for me ? 

Obviosly telemetry data coming into interface should make sense to all consumers and not to be custom fit for one specific implementation.

It just one specific example of a more general leaking abstraction in design of IMetricTelemetryConsumer.  Orleans has many metrics constructed from other metrics but IMetricTelemetryConsumer don't have any general interface to implement this.

**Comment by galvesribeiro** [comment]
Hey guys, this PR is outdated with the lots of changes recently made by @jason-bragg 

Should we close it?

**Comment by jason-bragg** [comment]
Yeah, I've a work item to create a tracking issue for this and close it out.  I'll do that today.

**Comment by jason-bragg** [comment]
Closing as alternative solutions are preferred and PR is out of date.
Added tracking issue "Remove performance counters from ThreadTrackingStatistic #2609"

**Pull request closed without merging**